pkg/backend/display/object_diff.go

```diff
@@ -329,6 +329,9 @@ func massageStackPreviewOutputDiff(diff *resource.ObjectDiff, inResource bool) {
 			delete(diff.Updates, k)
 		} else {
 			massageStackPreviewDiff(d, inResource)
+			if !d.Object.AnyChanges() {
+				delete(diff.Updates, k)
+			}
```

Comment:
Here's the patch: `massageStackPreviewDiff(…)` removes the `@isPulumiInstance` property, but `d` is still not `nil`, so the `display` treats it as if it has an update, even though it doesn't.

Several code paths check if the Diff `== nil`, and I suppose changing those checks to look for `.AnyChanges()` would also fix the issue, but I was unable to find a neat way to make that work.

(This is the reason I filed #12256 yesterday)